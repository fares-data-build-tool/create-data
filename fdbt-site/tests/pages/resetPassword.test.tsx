import * as React from 'react';
import { shallow } from 'enzyme';
import ResetPassword from '../../src/pages/resetPassword';

describe('resetPassword', () => {
    let expiryDate: number;

    beforeEach(() => {
        expiryDate = Math.abs(new Date(2020, 5, 30).getUTCSeconds());
    });

    it('should render correctly', () => {
        const tree = shallow(
            <ResetPassword
                username="test@tfn.com"
                regKey="123456"
                expiry={expiryDate.toString()}
                csrfToken=""
                errors={[]}
            />,
        );
        expect(tree).toMatchSnapshot();
    });

    it('should render error messaging when errors are passed', () => {
        const tree = shallow(
            <ResetPassword
                username="test@tfn.com"
                regKey="123456"
                expiry={expiryDate.toString()}
                csrfToken=""
                errors={[
                    {
                        errorMessage: 'Passwords do not match',
                        id: 'password',
                    },
                ]}
            />,
        );
        expect(tree).toMatchSnapshot();
    });
});
