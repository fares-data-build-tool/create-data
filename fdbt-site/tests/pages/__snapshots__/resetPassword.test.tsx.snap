// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetPassword should render correctly 1`] = `
<Component
  description="Reset Password page of the Fares data build tool"
  errors={Array []}
  title="Reset Password - Fares data build tool"
>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-two-thirds"
    >
      <CsrfForm
        action="/api/resetPassword"
        csrfToken=""
        method="post"
      >
        <ErrorSummary
          errors={Array []}
        />
        <div
          className="govuk-form-group "
        >
          <fieldset
            aria-describedby="reset-password-page-heading"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--l"
            >
              <h1
                className="govuk-fieldset__heading"
                id="reset-password-page-heading"
              >
                Reset your password
              </h1>
            </legend>
            <p
              className="govuk-hint hint-text"
              id="reset-password-page-hint"
            >
              Your password should be at least 8 characters long
            </p>
            <div
              className="govuk-form-group"
            >
              <label
                className="govuk-label"
                htmlFor="new-password"
              >
                New password
              </label>
              <FormElementWrapper
                errorClass="govuk-input--error"
                errorId="new-password"
                errors={Array []}
              >
                <input
                  aria-describedby="reset-password-page-hint"
                  className="govuk-input"
                  id="new-password"
                  name="password"
                  spellCheck="false"
                  type="password"
                />
              </FormElementWrapper>
            </div>
            <div
              className="govuk-form-group"
            >
              <label
                className="govuk-label"
                htmlFor="confirm-new-password"
              >
                Confirm password
              </label>
              <FormElementWrapper
                errorClass="govuk-input--error"
                errorId="confirm-new-password"
                errors={Array []}
              >
                <input
                  aria-describedby="reset-password-page-hint"
                  className="govuk-input"
                  id="confirm-new-password"
                  name="confirmPassword"
                  spellCheck="false"
                  type="password"
                />
              </FormElementWrapper>
            </div>
          </fieldset>
        </div>
        <input
          className="govuk-button"
          id="reset-password-button"
          name="resetPassword"
          type="submit"
          value="Reset Password"
        />
        <input
          name="regKey"
          type="hidden"
          value="123456"
        />
        <input
          name="username"
          type="hidden"
          value="test@tfn.com"
        />
        <input
          name="expiry"
          type="hidden"
          value="0"
        />
      </CsrfForm>
    </div>
  </div>
</Component>
`;

exports[`resetPassword should render error messaging when errors are passed 1`] = `
<Component
  description="Reset Password page of the Fares data build tool"
  errors={
    Array [
      Object {
        "errorMessage": "Passwords do not match",
        "id": "password",
      },
    ]
  }
  title="Reset Password - Fares data build tool"
>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-two-thirds"
    >
      <CsrfForm
        action="/api/resetPassword"
        csrfToken=""
        method="post"
      >
        <ErrorSummary
          errors={
            Array [
              Object {
                "errorMessage": "Passwords do not match",
                "id": "password",
              },
            ]
          }
        />
        <div
          className="govuk-form-group govuk-form-group--error"
        >
          <fieldset
            aria-describedby="reset-password-page-heading"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--l"
            >
              <h1
                className="govuk-fieldset__heading"
                id="reset-password-page-heading"
              >
                Reset your password
              </h1>
            </legend>
            <p
              className="govuk-hint hint-text"
              id="reset-password-page-hint"
            >
              Your password should be at least 8 characters long
            </p>
            <div
              className="govuk-form-group"
            >
              <label
                className="govuk-label"
                htmlFor="new-password"
              >
                New password
              </label>
              <FormElementWrapper
                errorClass="govuk-input--error"
                errorId="new-password"
                errors={
                  Array [
                    Object {
                      "errorMessage": "Passwords do not match",
                      "id": "password",
                    },
                  ]
                }
              >
                <input
                  aria-describedby="reset-password-page-hint"
                  className="govuk-input"
                  id="new-password"
                  name="password"
                  spellCheck="false"
                  type="password"
                />
              </FormElementWrapper>
            </div>
            <div
              className="govuk-form-group"
            >
              <label
                className="govuk-label"
                htmlFor="confirm-new-password"
              >
                Confirm password
              </label>
              <FormElementWrapper
                errorClass="govuk-input--error"
                errorId="confirm-new-password"
                errors={
                  Array [
                    Object {
                      "errorMessage": "Passwords do not match",
                      "id": "password",
                    },
                  ]
                }
              >
                <input
                  aria-describedby="reset-password-page-hint"
                  className="govuk-input"
                  id="confirm-new-password"
                  name="confirmPassword"
                  spellCheck="false"
                  type="password"
                />
              </FormElementWrapper>
            </div>
          </fieldset>
        </div>
        <input
          className="govuk-button"
          id="reset-password-button"
          name="resetPassword"
          type="submit"
          value="Reset Password"
        />
        <input
          name="regKey"
          type="hidden"
          value="123456"
        />
        <input
          name="username"
          type="hidden"
          value="test@tfn.com"
        />
        <input
          name="expiry"
          type="hidden"
          value="0"
        />
      </CsrfForm>
    </div>
  </div>
</Component>
`;
